File: packages/sdk-typescript/examples/chat-workflow.ts
import { ArcEnCielClient } from '../src/index.js'

const environment = (globalThis as typeof globalThis & { process?: { env?: Record<string, string | undefined> } }).process?.env ?? {}
const apiKey = environment.ARCENCIEL_API_KEY
const threadId = Number(environment.ARCENCIEL_CHAT_THREAD_ID ?? 0)

async function main(): Promise<void> {
  if (!apiKey) throw new Error('ARCENCIEL_API_KEY is required')

  const client = new ArcEnCielClient({ apiKey })
  const inbox = await client.call(() => client.chat.listChatThreads({ folder: 'inbox', limit: 10 }))
  console.log(inbox.data.map(thread => ({ id: thread.id, title: thread.title, unread: thread.hasUnread })))

  if (threadId > 0) {
    const created = await client.call(() =>
      client.chat.createChatMessage({
        threadId,
        idempotencyKey: crypto.randomUUID(),
        content: 'SDK v1.7 chat example',
      })
    )
    console.log({ createdMessageId: created.id, requestIdHint: 'Inspect ArcEnCielError.requestId on failures.' })
  }
}

void main()

File: packages/sdk-python/examples/chat_workflow.py
"""Run with ARCENCIEL_API_KEY and optionally ARCENCIEL_CHAT_THREAD_ID."""

from __future__ import annotations

import os
from uuid import uuid4

from arcenciel import ArcEnCielClient

api_key = os.environ["ARCENCIEL_API_KEY"]
thread_id = int(os.environ.get("ARCENCIEL_CHAT_THREAD_ID", "0"))
client = ArcEnCielClient(api_key=api_key)

inbox = client.call_sync(lambda: client.chat.list_chat_threads_sync(folder="inbox", limit=10))
print(
    [{"id": thread.id, "title": thread.title, "unread": thread.has_unread} for thread in inbox.data]
)

if thread_id > 0:
    created = client.call_sync(
        lambda: client.chat.create_chat_message_sync(
            thread_id=thread_id,
            idempotency_key=str(uuid4()),
            content="SDK v1.7 chat example",
        )
    )
    print({"created_message_id": created.id})
